### PR TITLE
Story visualization slots

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fixes https://github.com/RasaHQ/rasa/issues/4132 (broken story visualization in Rasa X when stories had slots
- add more tests to io module
- add constant for default encoding (utf-8)
- add function `write_text_file` to `io` module
- move `dump_obj_as_json_to_file` to io module

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/trackers.py [hunk 1]
```diff
@@ -438,9 +438,9 @@ def export_stories(self, e2e=False) -> Text:
 
     def export_stories_to_file(self, export_path: Text = "debug.md") -> None:
         """Dump the tracker as a story to a file."""
+        from rasa.utils import io
```
Comment (tabergma) [inline rasa/core/trackers.py]:
```suggestion
        import rasa.utils.io
```

To keep it consistent.

File: rasa/core/trackers.py [hunk 2]
```diff
@@ -438,9 +438,9 @@ def export_stories(self, e2e=False) -> Text:
...
 
-        with open(export_path, "a", encoding="utf-8") as f:
-            f.write(self.export_stories() + "\n")
+        io.write_text_file(self.export_stories() + "\n", export_path, append=True)
```
Comment (tabergma) [inline rasa/core/trackers.py]:
```suggestion
        rasa.utils.io.write_text_file(self.export_stories() + "\n", export_path, append=True)
```

File: rasa/model.py [hunk 3]
```diff
@@ -269,7 +269,7 @@ def persist_fingerprint(output_path: Text, fingerprint: Fingerprint):
         fingerprint: The fingerprint to be persisted.
 
     """
-    from rasa.core.utils import dump_obj_as_json_to_file
+    from rasa.utils.io import dump_obj_as_json_to_file
```
Comment (tabergma) [inline rasa/model.py]: Aren't we avoiding direct imports?

File: rasa/utils/io.py [hunk 4]
```diff
@@ -194,17 +204,37 @@ def unarchive(byte_array: bytes, directory: Text) -> Text:
...
         yaml.dump(data, outfile, default_flow_style=False)
 
 
+def write_text_file(
+    content: Text,
+    file_path: Union[Text, Path],
+    encoding: Text = DEFAULT_ENCODING,
+    append: bool = False,
+) -> None:
+    """Writes text to  a file.
```
Comment (tabergma) [inline rasa/utils/io.py]:
```suggestion
    """Writes text to a file.
```

File: rasa/model.py (hunk 3 above)
Comment (wochinge) [reply rasa/model.py]: will change it, that was probably pycharms quick fix 👍

**Comment by wochinge** [comment]
> Just as a note: as quite a lot files changed, maybe consider to open a new PR for the additional stuff (e.g. moving stuff to io utils) to keep the PR small and clean. Also, the review can focus more on the actual change if you do so

Sorry, you're totally right! Will split it up the next time. 👍

**Pull request merged**